README.txt
Unity notifications

This project provides an implementation of the
Free Desktop Notification server for Unity. It also
provides some additional features as needed
on alternative platforms, such as mobile.

Since the Dash is implemented in QML, this functionality
is implemented as a QML plugin. Notification status is provided
as a QAbstractListModel, making integration simple. The exact
form of the API is defined by the Dash, Unity-notifications
only implements it.

Unity-notifications does not and will not provide a C/C++
API for sending or receiving notifications. It is QML only.

In the roadmap there is also plans to provide a QML plugin
for sending notifications from client applications.


debian/changelog
unity-notifications (0.1.4+ubports) xenial; urgency=medium

  * Imported to UBports

 -- UBports auto importer <infra@ubports.com>  Mon, 12 Mar 2018 07:12:31 +0000

unity-notifications (0.1.3+17.04.20161024-0ubuntu1) zesty; urgency=medium

  [ Lukáš Tinkl ]
  * Simplify the check for cmd line (notify-send) to make it work also
    inside a container (LP: #1623142)

 -- Michael Zanetti <michael.zanetti@canonical.com>  Mon, 24 Oct 2016 11:32:28 +0000

unity-notifications (0.1.3+16.10.20160819-0ubuntu1) yakkety; urgency=medium

  [ Lukáš Tinkl ]
  * Enable hardening security features (LP: #1613678)

 -- Michał Sawicz <michal.sawicz@canonical.com>  Fri, 19 Aug 2016 17:15:43 +0000

unity-notifications (0.1.3+16.10.20160809-0ubuntu1) yakkety; urgency=medium

  [ Lukáš Tinkl ]
  * Visible queue improvements (LP: #1606202)

 -- Michał Sawicz <michal.sawicz@canonical.com>  Tue, 09 Aug 2016 23:00:28 +0000

unity-notifications (0.1.3+15.10.20151021-0ubuntu1) wily; urgency=medium

  [ Michal Sawicz ]
  * Fix version number

  [ CI Train Bot ]
  * New rebuild forced.

  [ Lukáš Tinkl ]
  * Merge and rebase older code to fix notifications crashing on closing
    (LP: #1453958)

  [ Mirco Müller ]
  * Merge and rebase older code to fix notifications crashing on closing
    (LP: #1453958)

 -- Michał Sawicz <michal.sawicz@canonical.com>  Wed, 21 Oct 2015 11:48:57 +0000

unity-notifications (0.1.2+15.10.20150623.is.0.1.2+15.10.20150804-0ubuntu1) wily; urgency=medium

  [ Mirco Müller ]
  * Add main loop to loopless examples

 -- CI Train Bot <ci-train-bot@canonical.com>  Tue, 04 Aug 2015 12:15:26 +0000

unity-notifications (0.1.2+15.10.20150623.is.0.1.2+15.10.20150706.1-0ubuntu2~gcc5.1) wily; urgency=medium

  * No-change test rebuild for g++5 ABI transition

 -- Steve Langasek <steve.langasek@ubuntu.com>  Wed, 15 Jul 2015 16:23:37 +0000

unity-notifications (0.1.2+15.10.20150623.is.0.1.2+15.10.20150706.1-0ubuntu1) wily; urgency=medium

  [ CI Train Bot ]
  * New rebuild forced.

  [ Charles Kerr ]
  * When a notification is closed due to internals (e.g. due to
    timeout), ensure the NotificationClosed bus signal is emitted. (LP:
    #1470031)

 -- CI Train Bot <ci-train-bot@canonical.com>  Mon, 06 Jul 2015 18:14:35 +0000

unity-notifications (0.1.2+15.10.20150623-0ubuntu1) wily; urgency=medium

  [ Pete Woods ]
  * Add end to end DBus tests and use XML introspection data
  * Handle client death by cleaning up notifications
  * Handle reopening closed notifications
  * Notifications are now owned by the client that opened them

 -- CI Train Bot <ci-train-bot@canonical.com>  Tue, 23 Jun 2015 15:50:33 +0000

unity-notifications (0.1.2+15.04.20141104-0ubuntu1) vivid; urgency=low

  [ Mirco Müller ]
  * Added new hint for SwipeToAct-widget.

 -- Ubuntu daily release <ps-jenkins@lists.canonical.com>  Tue, 04 Nov 2014 13:03:34 +0000

unity-notifications (0.1.2+15.04.20141030-0ubuntu1) vivid; urgency=low

  [ Mirco Müller ]
  * Added support, examples and tests for synchronous/confirmation
    notifications. (LP: #1232633)

 -- Ubuntu daily release <ps-jenkins@lists.canonical.com>  Thu, 30 Oct 2014 10:59:34 +0000

unity-notifications (0.1.2+14.10.20140922-0ubuntu1) utopic; urgency=low

  [ Mirco Müller ]
  * Use a more robust filterText() and don't filter newline-characters.

 -- Ubuntu daily release <ps-jenkins@lists.canonical.com>  Mon, 22 Sep 2014 17:42:23 +0000

unity-notifications (0.1.2+14.10.20140918-0ubuntu1) utopic; urgency=low

  [ Mirco Müller ]
  * Added an internal text-filter to setSummary() and setBody()

 -- Ubuntu daily release <ps-jenkins@lists.canonical.com>  Thu, 18 Sep 2014 09:54:13 +0000

unity-notifications (0.1.2+14.10.20140907-0ubuntu1) utopic; urgency=low

  [ Mirco Müller ]
  * Renamed and added hints to expose the new button-tint hints for
    snap-decisions, which are needed to comply with the visual changes
    coming from Design (for RTM). Updated examples to reflect these
    changes. (LP: #1348092)

 -- Ubuntu daily release <ps-jenkins@lists.canonical.com>  Sun, 07 Sep 2014 19:46:29 +0000

unity-notifications (0.1.2+14.10.20140717-0ubuntu1) utopic; urgency=low

  [ Mirco Müller ]
  * Added close() q_invokable to allow notifications to be closed from
    the renderer (unity8) side. This facilitates fixing LP:1308011 (LP:
    #1308011)

 -- Ubuntu daily release <ps-jenkins@lists.canonical.com>  Thu, 17 Jul 2014 02:14:19 +0000

unity-notifications (0.1.2+14.10.20140623-0ubuntu1) utopic; urgency=low

  [ Mirco Müller ]
  * Changed maximum number of allowed actions to be 7. Updated related examples accordingly. 
  * Bump version to indicate support for maximum of 7 actions.
  * Added a handy example to more easily demonstrate the visual snap-
    decision queue of the notification-frontend.
  * Added the backend-part of sound-hint support with two additional
    examples demonstrating its use.

 -- Ubuntu daily release <ps-jenkins@lists.canonical.com>  Mon, 23 Jun 2014 09:32:14 +0000

unity-notifications (0.1.1+14.10.20140602-0ubuntu1) utopic; urgency=low

  [ Mirco Müller ]
  * Added a snap-decisions-specific hint to allow control over the
    timeout of a notification. (LP: #1295762)

 -- Ubuntu daily release <ps-jenkins@lists.canonical.com>  Mon, 02 Jun 2014 10:58:29 +0000

unity-notifications (0.1.1+14.10.20140514.1-0ubuntu1) utopic; urgency=low

  [ Antti Kaijanmäki ]
  * Call model.notificationUpdated() when notification updates.

 -- Ubuntu daily release <ps-jenkins@lists.canonical.com>  Wed, 14 May 2014 11:31:06 +0000

unity-notifications (0.1.1+14.10.20140505-0ubuntu1) utopic; urgency=low

  [ Mirco Müller ]
  * Make the Roles enum available to QML. (LP: #1285712)

 -- Ubuntu daily release <ps-jenkins@lists.canonical.com>  Mon, 05 May 2014 12:25:31 +0000

unity-notifications (0.1.1+14.04.20140402-0ubuntu1) trusty; urgency=low

  [ Mirco Müller ]
  * To avoid false expectations the examples, using not yet re-
    implemented hints or features, were moved to the unsupported

examples/example.py
#!/usr/bin/python

################################################################################
##3456789 123456789 123456789 123456789 123456789 123456789 123456789 123456789
##      10        20        30        40        50        60        70        80
##
## Info: 
##    Example of how to use libnotify correctly and at the same time comply to
##    the new jaunty notification spec (read: visual guidelines)
##
## Run:
##    chmod +x example.py
##    ./example.py
##
## Copyright 2012 Canonical Ltd.
##
## Author:
##    Mirco "MacSlow" Mueller <mirco.mueller@canonical.com>
##
## This program is free software: you can redistribute it and/or modify it
## under the terms of the GNU General Public License version 3, as published
## by the Free Software Foundation.
##
## This program is distributed in the hope that it will be useful, but
## WITHOUT ANY WARRANTY; without even the implied warranties of
## MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR
## PURPOSE.  See the GNU General Public License for more details.
##
## You should have received a copy of the GNU General Public License along
## with this program.  If not, see <http://www.gnu.org/licenses/>.
##
################################################################################

import pynotify

# even in Python this is globally nasty :), do something nicer in your own code
capabilities = {'actions':                              False,
		'body':                                 False,
		'body-hyperlinks':                      False,
		'body-images':                          False,
		'body-markup':                          False,
		'icon-multi':                           False,
		'icon-static':                          False,
		'sound':                                False,
		'sound-file':                           False,
		'suppress-sound':                       False,
		'image/svg+xml':                        False,
		'urgency':                              False,
		'value':                                False,
		'x-canonical-value-bar-tint':           False,
		'x-canonical-private-synchronous':      False,
		'x-canonical-private-icon-only':        False,
		'x-canonical-truncation':               False,
		'x-canonical-snap-decisions':           False,
		'x-canonical-snap-decisions-timeout':   False,
		'x-canonical-snap-decisions-swipe':     False,
		'x-canonical-switch-to-application':    False,
		'x-canonical-secondary-icon':           False,
		'x-canonical-private-affirmative-tint': False,
		'x-canonical-private-rejection-tint':   False,
		'x-canonical-private-menu-model':       False,
		'x-canonical-non-shaped-icon':          False}

def initCaps ():
	caps = pynotify.get_server_caps ()
	if caps is None:
	        print "Failed to receive server caps."
		sys.exit (1)

	for cap in caps:
		capabilities[cap] = True

def printCaps ():
	info = pynotify.get_server_info ()
	print "Name:          " + info["name"]
	print "Vendor:        " + info["vendor"]
	print "Version:       " + info["version"]
	print "Spec. Version: " + info["spec-version"]

	caps = pynotify.get_server_caps ()
	if caps is None:
	        print "Failed to receive server caps."
		sys.exit (1)

	print "Supported capabilities/hints:"
	if capabilities['actions']:
		print "\tactions"
	if capabilities['body']:
		print "\tbody"
	if capabilities['body-hyperlinks']:
		print "\tbody-hyperlinks"
	if capabilities['body-images']:
		print "\tbody-images"
	if capabilities['body-markup']:
		print "\tbody-markup"
	if capabilities['icon-multi']:
		print "\ticon-multi"
	if capabilities['icon-static']:
		print "\ticon-static"
	if capabilities['sound']:
		print "\tsound"
	if capabilities['sound-file']:
		print "\tsound-file"
	if capabilities['suppress-sound']:
		print "\tsuppress-sound"
	if capabilities['urgency']:
		print "\turgency"
	if capabilities['value']:
		print "\tvalue"
	if capabilities['x-canonical-value-bar-tint']:
		print "\tx-canonical-value-bar-tint"
	if capabilities['image/svg+xml']:
		print "\timage/svg+xml"
	if capabilities['x-canonical-private-synchronous']:
		print "\tx-canonical-private-synchronous"
	if capabilities['x-canonical-private-icon-only']:
		print "\tx-canonical-private-icon-only"
	if capabilities['x-canonical-truncation']:
		print "\tx-canonical-truncation"
	if capabilities['x-canonical-snap-decisions']:
		print "\tx-canonical-snap-decisions"
	if capabilities['x-canonical-snap-decisions-timeout']:
		print "\tx-canonical-snap-decisions-timeout"
	if capabilities['x-canonical-snap-decisions-swipe']:
		print "\tx-canonical-snap-decisions-swipe"
	if capabilities['x-canonical-switch-to-application']:
		print "\tx-canonical-switch-to-application"
	if capabilities['x-canonical-secondary-icon']:
		print "\tx-canonical-secondary-icon"
	if capabilities['x-canonical-private-affirmative-tint']:
		print "\tx-canonical-private-affirmative-tint"
	if capabilities['x-canonical-private-rejection-tint']:
		print "\tx-canonical-private-rejection-tint"
	if capabilities['x-canonical-private-menu-model']:
		print "\tx-canonical-private-menu-model"
	if capabilities['x-canonical-non-shaped-icon']:
		print "\tx-canonical-non-shaped-icon"

	print "Notes:"
	if info["name"] == "notify-osd":
		print "\tx- and y-coordinates hints are ignored"
		print "\texpire-timeout is ignored"
		print "\tbody-markup is accepted but filtered"
	else:
		print "\tnone"

def closedHandler (notification, loop):
	print "\"closed\"-handler called"
	loop.quit ()
